lib/promoted/ruby/client/options.rb

```diff
@@ -166,25 +168,25 @@ def request_params include_insertion: true
         end
 
         def compact_insertions
-          @compact_insertions = []
-          insertions_to_compact = full_insertion
-          paging = request[:paging] || {}
-          size = paging[:size]
-          unless size.nil? || size == 0
-            insertions_to_compact = insertions_to_compact[0..size-1]
-          end
-          from = paging[:from].to_i
+          @insertion            = [] # insertion should be set according to the compact insertion
+          paging                = request[:paging] || {}
+          size                  = paging[:size] ? paging[:size].to_i : 0
+          from                  = paging[:from].to_i
+          insertions_to_compact = full_insertion[from..size-1]
```

Comment:
I think we'll need an argument that indicates the paging type:
- pre-trimmed - then we want to use `[0..size-1]`
- not-trimmed - then we want to use `[offset..size-1]`

This is an issue with the promoted-ts-client too.

@scottmcmaster - brought up this issue earlier today.

Reply:
For our first customer and this first case, we will probably just get the paginated results (pre-trimmed).

When we integrate Delivery API with them, we'll get the full list (not-trimmed).

Reply:
@danbosnichill if we aren't going to spec that out now and do it, I wonder if for now we should revert this change to go from 0 rather than "from/offset"?